### DISCUSSION TITLE
Catch exception `DagsterInvalidDefinitionError`

### DISCUSSION THREAD

**Discussion by kennycontreras** [dagster-io/dagster#21330] [General]
Hello everyone 👋 

I'm working on applying some minor improvements to a Dagster environment, and I'd like to be able to catch an exception when Dagster, for example, finds a duplicated job name. By default, this will raise `DagsterInvalidDefinitionError` 

I've been trying to catch the exception without luck. 

Here's an example of my `repo.py` file to load the pipelines

```python    

  if "py" in file:
      try:
          # code to add the pipelines to the Dagster environment... 
          ##
          ####
      except (ImportError, TypeError, ModuleNotFoundError, NameError, DagsterInvalidDefinitionError):
          logging.exception(f"Failed importing: {file}")
          continue
```

All the other exceptions work but not `DagsterInvalidDefinitionError`. 
I've tried importing `DagsterInvalidDefinitionError` from the Dagster library and even from the `_core` module.

Does anyone have any tips or has tried something similar? 

Thanks!

**Comment by garethbrickman** [comment]
A couple things to troubleshoot:

1. Ensure that you are importing `DagsterInvalidDefinitionError` from dagster.core.errors.
2. Verify that the code which might raise the `DagsterInvalidDefinitionError` is indeed within the try block.
3. Add a generic except Exception as e clause to your try block to catch all exceptions and log them. This can help you determine if the `DagsterInvalidDefinitionError` is being raised at all and if it is being caught by the more generic Exception class.
